Dockerfile

```diff
@@ -0,0 +1,164 @@
+#       -e RAILS_ENV=production -e SECRET_TOKEN=foobar opf/openproject
+#
+# The worker process can be run in a separate container.
+#
+#     docker run -d --link postgres:database \
+#       -e DATABASE_URL=postgres://openproject:openproject@database:5432/openproject \
+#       -e RAILS_ENV=production -e SECRET_TOKEN=foobar opf/openproject \
+#       bundle exec rake jobs:work
+#
+# If you don't want to execute these commands by hand all the time, have a look at the fig.yml
+
+FROM ubuntu:14.04
+MAINTAINER Finn GmbH <info@finn.de>
+
+# update the package list
+RUN apt-get update
+
+# we use a database for our backend, so let's add a couple of them
+RUN apt-get -y install libpq-dev libmysql++-dev sqlite3 libsqlite3-dev
+
+# newer versions of openproject use angular-js, so we want node and npm
+RUN apt-get -y install nodejs npm
+
+# allow the nodejs binary to be called via node (this should be fixed somehow)
+RUN ln -s nodejs /usr/bin/node
+
+# install git in order to checkout gems defined by git urls
+RUN apt-get -y install git-core
+
+# install ruby 2.1
+#
+# the following installs ruby 2.1 via the apt package manager. unfortunately,
+# the default packages still only contain ruby 1.9, so we use brightbox's one.
+#
+RUN apt-get install -y software-properties-common python-software-properties
+RUN apt-add-repository -y ppa:brightbox/ruby-ng
+RUN apt-get update
+
+# install ruby 2.1 with development extensions
+RUN apt-get install -y ruby2.1 ruby2.1-dev
+
+# install our beloved bundler
+RUN gem install bundler
+
+# add openproject's source to the container
+COPY . /usr/src/openproject
+
+# set the working directory to the app
+WORKDIR /usr/src/openproject
+
+# add two gems that are required to make everything work inside a container
+#
+# sqlite3: for connecting to a dummy database during asset precompilation
+# 12factor: to serve static assets by rails and log to stdout in production
+#
+RUN echo "gem 'sqlite3', '~> 1.3.0'" >> ./Gemfile.local
+RUN echo "gem 'rails_12factor', '~> 0.0.3'" >> ./Gemfile.local
+
+# create an unprivileged user that owns the code and will run the app server
+RUN useradd -m openproject
+RUN chown -R openproject:openproject /usr/src/openproject
+
+# switch to the openproject user from now on
+USER openproject
+
+# install all ruby dependencies via bundler into a local path
+RUN bundle install --path ./vendor/bundle
+
+# install all node dependencies via npm
+RUN npm install
+
+# use a database.yml that takes its values from an environment variable
+RUN cp ./config/database.env.yml ./config/database.yml
+
+# precompile all assets
+#
+# openproject currently requires a database when compiling assets
+# we fake one by providing an sqlite db location for this command only
+#
+# note, asset precompilation needs to be done in production, otherwise icon images are missing
+#
+RUN RAILS_ENV=production SECRET_TOKEN=foobar DATABASE_URL=sqlite3://db/ignore_me.sqlite3 \
```

Comment:
i thing the fake token for the precompilation step is fine. it's only necessary because precompilation needs to be done in production. running `rake generate_secret_token` before would allow us to skip defining this env var here but would also include it in the image which is probably not what you want.